### PR TITLE
adds other teaching periods confirmation page to cfp journey

### PR DESCRIPTION
**Pull request by J05h-L**
### Context
Story - [AB#191593](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/191593)
Tasks - [AB#198910](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/198910) & [AB#198911](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/198911)

### Change proposed in this pull request
Adds the confirmation page to proceed without adding other teaching periods to the plan for the cfp journey
adds validation if the users does not select yes or no

### Guidance to review 
Hint text used rather than a warning text to match the style used in previous steps of the journey.
Title also changed from the ticket for the same reasons

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace EducationBenchmarking.Web.Validators.FinancialPlanStages
+{
+    public class OtherTeachingPeriodsConfirmStageValidator : AbstractValidator<OtherTeachingPeriodsConfirmStage>
+    {
+        public OtherTeachingPeriodsConfirmStageValidator()
+        {
+            RuleFor(p => p.Proceed)
+                .Must(x => x.HasValue)
+                .WithMessage("Please select yes or no")
+                .WithName("Proceed");
```
Comment (jrabbott) [inline web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs:13]: This is not require,

File: web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace EducationBenchmarking.Web.Validators.FinancialPlanStages
+{
+    public class OtherTeachingPeriodsConfirmStageValidator : AbstractValidator<OtherTeachingPeriodsConfirmStage>
+    {
+        public OtherTeachingPeriodsConfirmStageValidator()
+        {
+            RuleFor(p => p.Proceed)
+                .Must(x => x.HasValue)
+                .WithMessage("Please select yes or no")
```
Comment (jrabbott) [inline web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs:12]: Can we update the message to follow design system guidance?

Select yes if [whatever it is is true]

https://design-system.service.gov.uk/components/radios/#error-messages

File: web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs (hunk 1 above)
Comment (J05h-L) [reply web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs]: Removed in latest commit

File: web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs (hunk 2 above)
Comment (J05h-L) [reply web/src/EducationBenchmarking.Web/Validators/FinancialPlanStages/OtherTeachingPeriodsConfirmStageValidator.cs]: Updated in latest commit

**Review by jrabbott — APPROVED** [review]

**Pull request merged**